Cargo.toml

```diff
@@ -11,4 +12,13 @@ edition = "2018"
 
 [dependencies]
 async-trait = "0.1"
-futures = "0.3"
+tokio = { version = "0.2", features = ["fs", "rt-threaded", "macros"] }
+futures = { version = "0.3", optional = true }
```

Comment:
Again, we don't have to pull in whole futures crate. We can only use `futures-util` crate (which is a sub-crate of `futures`) with `io` feature enabled to minimize the depndency tree.

Reply:
done